build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -108,12 +109,126 @@ service Execution {
     option (google.api.http) = { post: "/v2/{instance_name=**}/actions:execute" body: "*" };
   }
 
+  // Executes an action remotely, allowing dynamic retrieval of directory and
+  // file inputs from the client where the client did not upload the inputs
+  // beforehand. The client must remain connected to the worker during the
+  // duration of the action execution.
+  //
+  // This call may not be implemented on all platforms. Its support is
+  // indicated by
+  // [ExecutionCapabilities.dynamic_execution_support][build.bazel.remote.execution.v2.ExecutionCapabilities.dynamic_execution_support].
+  //
+  // In order to execute an action, the client must first upload the best
+  // guesses for the action's file inputs (which can be zero files if
+  // prediction is unavailable), the
+  // [Command][build.bazel.remote.execution.v2.Command] to run, and the
+  // [Action][build.bazel.remote.execution.v2.Action] containing the best
+  // guess for the `Action`'s needed directory and file layout into the
+  // [ContentAddressableStorage][build.bazel.remote.execution.v2.ContentAddressableStorage].
+  // It then calls `ExecuteWithDynamicRetrieval` with an `action_digest`
+  // referring to them. The worker will run the action and eventually return
+  // the result.
+  //
+  // The `Action.input_root_digest` directory tree provides a filesystem
+  // layout that may be underspecified in terms of missing directories or
+  // unspecified file hashes. Any `Directory` entry provided in this tree
+  // must contain at least the subdirectory names and file names and sizes
+  // needed to enumerate the complete contents of the directory.
+  //
+  // A typical implementation of a server using dynamic execution will use a
+  // virtual filesystem to allow receiving callbacks about files and
+  // directories requested via operating system filesytem APIs. The server
+  // MAY allow rendering of a fully qualified, rooted filesystem by the
+  // client, by accepting in `Action.input_root_digest` a `Directory` tree
+  // that contains a filesystem root appropriate to the operating system.
+  // By convention, representation of such a filesystem utilizes a Directory
+  // with an empty string for a name. Under Linux this translates into the
```

Comment:
This has significant security implications as it could allow a malicious worker to exfiltrate arbitrary information, including private keys and the like, from the client's filesystem. There should be some discussion of this.